pkg/sql/catalog/accessor.go

```diff
@@ -40,11 +40,11 @@ type Accessor interface {
 		ctx context.Context, txn *kv.Txn, dbName string, flags tree.DatabaseLookupFlags,
 	) (DatabaseDescriptor, error)
 
-	// GetSchema returns true and a ResolvedSchema object if the target schema
+	// GetSchema returns true and a SchemaDescriptor object if the target schema
 	// exists under the target database.
 	GetSchemaByName(
 		ctx context.Context, txn *kv.Txn, dbID descpb.ID, scName string, flags tree.SchemaLookupFlags,
-	) (bool, ResolvedSchema, error)
+	) (bool, SchemaDescriptor, error)
```

Comment:
Considering that we return an interface instead of struct now, should we do away with the bool return field, since it is equivalent to a nil check? 

I know there are two schools of thought on this matter.  Do we have a consensus on this at CRL? If so I haven't noticed. Personally I lean slightly towards nil-checking for a bunch of debatable reasons which aren't really super interesting but honestly I don't care too much either way.